File: BudgetServices/BudgetServices/DTO/ExpenseDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace BudgetServices.DTO
{
    public class ExpenseDTO
    {
        public string ExpenseID = null;
        public DateTime ExpenseDate = new DateTime(1900, 1, 1);
        public string ExpenseName = "";
        public string ExpenseType = "";

        public ExpenseDTO(string expenseID, DateTime? expenseDate = null, string expenseName = null, string expenseType = null)
        {
            this.ExpenseID = expenseID;

            if (expenseDate != null) 
                this.ExpenseDate = (DateTime)expenseDate;

            if (expenseName != null)
                this.ExpenseName = expenseName;

            if (expenseType != null)
                this.ExpenseType = expenseType;
        }
    }
}

File: budget/src/presentation/dto/expense-dto.ts
export class ExpenseDTO {
    public ExpenseID: string = null;
    public ExpenseDate: Date = new Date(1900, 1, 1);
    public ExpenseName: string = '';
    public ExpenseType: string = '';
    
    constructor(expenseID?: string, expenseDate?: Date, expenseName?: string, expenseType?: string) {
        this.ExpenseID = expenseID;
        this.ExpenseDate = expenseDate;
        this.ExpenseName = expenseName;
        this.ExpenseType = expenseType;
    }

    static Default(): ExpenseDTO {
        return new ExpenseDTO(null, new Date(1900, 1, 1), '', '');
    }
}
